lib/rucio/transfertool/fts3.py

```diff
@@ -399,16 +436,36 @@ def build_job_params(
         elif 'default' in max_time_in_queue:
             job_params['max_time_in_queue'] = max_time_in_queue['default']
 
-    overwrite_hop = True
-    for transfer_hop in transfer_path[:-1]:
-        # Only allow overwrite if all hops in multihop allow it
-        h_overwrite = transfer_hop.rws.attributes.get('overwrite', True)
-        job_params['overwrite'] = h_overwrite and job_params['overwrite']
-        # Allow overwrite_hop if all intermediate hops allow it (ignoring the last hop)
-        overwrite_hop = h_overwrite and overwrite_hop
+    # Refer to https://its.cern.ch/jira/browse/FTS-1749 for full details (login needed), extract below:
+    # Why the overwrite_hop parameter is needed?
+    # Rucio decides that a multihop transfer is needed DISK1 --> DISK2 --> TAPE1 in order to put the file on tape. For some reason, the file already exists on DISK2.
+    # Rucio doesn't know about this file on DISK2. It could be either a correct or corrupted file. This can be due to a previous issue on Rucio side, FTS side, network side, etc (many possible reasons).
+    # Normally, Rucio allows overwrite towards any disk destination, but denies overwrite towards a tape destination. However, in this case, because the destination of the multihop is a tape, DISK2 cannot be overwritten.
+    # Proposed solution
+    # Provide an --overwrite-hop submission option, which instructs FTS to overwrite all transfers except for the destination within a multihop submission.
+
+    # direct transfers, is not multihop
+    if len(transfer_path) == 1:
+        overwrite_hop = False
+
+    else:
+        overwrite_hop = True
+        for transfer_hop in transfer_path[:-1]:
+            # Only allow overwrite if all hops in multihop allow it
+            h_overwrite = transfer_hop.rws.attributes.get('overwrite', True)
+
+            # if one hop cannot be overwritten, quit early
+            if h_overwrite is False:
+                break
+
+            job_params['overwrite'] = h_overwrite and job_params['overwrite']
+            # Allow overwrite_hop if all intermediate hops allow it (ignoring the last hop)
+            overwrite_hop = h_overwrite and overwrite_hop
+
```

Comment:
```suggestion
            job_params['overwrite'] = h_overwrite and job_params['overwrite']
            # Allow overwrite_hop if all intermediate hops allow it (ignoring the last hop)
            overwrite_hop = h_overwrite and overwrite_hop
            if h_overwrite is False:
                break
```

The `break` needs to be after the code block at 461-463, otherwise `job_params['overwrite']` and ` overwrite_hop` might remain `True` even though `h_overwrite` is `False`.

I'm more in favour of the `all()` approach:

```
# Set `overwrite` to `True` only if all hops allow it
job_params['overwrite'] = overwrite_hop = all(transfer_hop.rws.attributes.get('overwrite', True) for transfer_hop in transfer_path[:-1])
```

Since it clarifies that all hops must allow overwriting, and it breaks early if one of them returns `False`

Reply:
Makes sense and is cleaner code